src/rkr/tracing/Thread.cc

```diff
@@ -1511,6 +1511,29 @@ void Thread::_socket(Build& build,
                      int protocol) noexcept {
   WARN << "socket(2) not yet implemented. Emulating as an anonymous file.";
 
+  //if (!options::frontier) {
+  //  WARN << "IN FRONTIER MODE, SOCKET CALLED, EXITTING";
+  //  std::exit(159);
+  //}
+
+  finishSyscall([=](Build& build, const IRSource& source, long rc) {
+    resume();
+
+    if (rc >= 0) {
+      auto ref = getCommand()->nextRef();
+      build.fileRef(source, getCommand(), 0600, ref);
+      bool cloexec = (type & SOCK_CLOEXEC) == SOCK_CLOEXEC;
+      _process->addFD(build, source, rc, ref, cloexec);
```

Comment:
Quick question: Is this code similar to what is done in other syscalls? If it contains something special could you add a comment describing it?

Reply:
yeah, socket was already implemented in riker, https://github.com/curtsinger-lab/riker/blob/44aa5a0b469c43d943730294dc1a04d5d8ebba9c/src/rkr/tracing/Thread.cc#L1514